compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -641,30 +641,21 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
 
                         if Some(trait_ref.def_id()) == self.tcx.lang_items().sized_trait() {
                             if !self.tcx.object_safety_violations(trait_ref.def_id()).is_empty() {
-                                match (
-                                    obligation.cause.code(),
-                                    trait_ref.skip_binder().self_ty().kind(),
-                                ) {
-                                    (
-                                        ObligationCauseCode::WhereClauseInExpr(..),
-                                        ty::Dynamic(..),
-                                    ) => {
+                                if let ObligationCauseCode::SizedCallReturnType(did)
```

Comment:
Right, I've just arrived at this commit (the 4th). Noticing you had to make this change less "aggressive" makes me think that this branch is still super fragile. Like I don't know if the `delay_as_bug` is correct in all perceivable cases

Reply:
All `delay_as_bug`s are now gone after the "Fix tidy" commit.